EIPS/eip-5749.md

```diff
@@ -0,0 +1,118 @@
+
+
+Replacing `window.ethereum` with `window.evmproviders` will allow solutions such as web3modal and web3onboard to only display all injected wallets the user has installed. This will simpify the UX and remove race conditions between wallet providers in case multiple wallets are installed. Over time, as `window.evmproviders` supplants the current standard and removes barriers to choice, we can hope to see a wallet landscape more reflective of user preference.
+
+
+
+### Some wallets that current;y use `window.ethereum`
+	Metamask
+	Coinbase wallet
+	Enkrypt
+	Rabby
+	Trust wallet
+	Rainbow ..etc
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### `window.evmproviders={}`
+
+
+```typescript
+interface ProviderInfo {
+	name: string
+	icon:  `data:image/svg+xml;base64,${string}`
+	description: string
+}
+interface ProviderWithInfo extends EIP1193Provider {
+	info: ProviderInfo
+}
+interface EVMProvidersType {
+	[key: string]: ProviderWithInfo;
+}
+interface  Window {
+	evmproviders: EVMProvidersType
+}
+```
+Type `EIP1193Provider` is well documented at [EIP-1193](./eip-1193.md)
+
+```
+interface ProviderInfo
+
+name: Name of the Wallet
+icon: base64 encoded svg image
+description: Description for your wallet
+```
+
+```
+interface EVMProvidersType
+key is RECOMMENDED to be the name of the extension
+```
+By adopting an object for EIP-1193 compliant providers we can have multiple different ethereum/evm compatible wallets coexists in the same browser. This will prevent race conditions and inconsistent behaviors.
+
+By introducing `ProviderInfo` type web onboarding libraries such as 
+	Web3Modal
+	Web3React
+	Web3Onboard
+can easily grab the necessary information to populate their popup window to choose the wallet. 
+
+
+
+## Rationale
+name (`evmproviders`)  was chosen in order to be inclusive of all other chains that are evm and EIP-1193 compliant. 
+**svg** was chosen since it is easier to be modified if the application requires for example different size for the image. 
+
+## Backwards Compatibility
+This EIP is not intended to replace current behavior for `window.ethereum`. Therefore it SHOULD NOT break existing applications. However, the goal is to adopt this standard over time and deprecate `window.ethereum` and make `window.evmproviders` the RECOMMENDED way of injecting browser wallets. 
+
+
+## Reference Implementation
+### injection
+```	
+	const provider: ProviderWithInfo = [your wallet]
+	window.evmproviders = window.evmproviders || {};
+	window.evmproviders[name] = provider
```

Comment:
this is not happening in the content script, this is part of the injection where currently `ethereum` object gets injected to global window. instead of injecting `ethereum` this eip will inject `evmproviders={}`